ntbs-service/Models/Validations/NhsNumberValidationAttribute.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Text.RegularExpressions;
+
+namespace ntbs_service.Models.Validations
+{
+    public class ValidNhsNumberAttribute : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            if(value == null)
+            {
+                return null;
+            }
+            var nhsNumber = value.ToString();
+            var match = Regex.Match(nhsNumber, "^[0-9]+$");
+            if(!match.Success)
+            {
+                return new ValidationResult(ValidationMessages.NhsNumberFormat);
+            }
+            else if(nhsNumber.Length != 10)
+            {
+                return new ValidationResult(ValidationMessages.NhsNumberLength);
+            }
+            
+            var firstDigit = nhsNumber.Substring(0, 1);
+            var  scottishAndTestDigits = new List<string> {"0", "1", "2", "9"};
+            if (scottishAndTestDigits.Contains(firstDigit))
+            {
+                return null;

```

Comment:
This deserves a quick comment, as it's not obvious why Scottish numbers wouldn't need validation. It's also worth pointing out that 9 is a generally-used test digit, not something that we've added ourselves!